pyzebra/load_1D.py

```diff
@@ -0,0 +1,212 @@
+    ("omega_angle", float),
+    ("chi_angle", float),
+    ("phi_angle", float),
+)
+
+CCL_FIRST_LINE_NB = (
+    *CCL_FIRST_LINE,
+    ("gamma_angle", float),
+    ("omega_angle", float),
+    ("nu_angle", float),
+    ("unkwn_angle", float),
+)
+
+CCL_SECOND_LINE = (
+    ("number_of_measurements", int),
+    ("angle_step", float),
+    ("monitor", float),
+    ("temperature", float),
+    ("mag_field", float),
+    ("date", str),
+    ("time", str),
+    ("scan_type", str),
+)
+
+
+def load_1D(filepath):
+    """
+    Loads *.ccl or *.dat file (Distinguishes them based on last 3 chars in string of filepath
+    to add more variables to read, extend the elif list
+    the file must include '#data' and number of points in right place to work properly
+
+    :arg filepath
+    :returns det_variables
+    - dictionary of all detector/scan variables and dictinionary for every measurement.
+    Names of these dictionaries are M + measurement number. They include HKL indeces, angles,
+    monitors, stepsize and array of counts
+    """
+    with open(filepath, "r") as infile:
+        _, ext = os.path.splitext(filepath)
+        det_variables = parse_1D(infile, data_type=ext)
+
+    return det_variables
+
+
+def parse_1D(fileobj, data_type):
+    # read metadata
+    metadata = {}
+    for line in fileobj:
+        if "=" in line:
+            variable, value = line.split("=")
+            variable = variable.strip()
+            if variable in META_VARS_FLOAT:
+                metadata[variable] = float(value)
+            elif variable in META_VARS_STR:
+                metadata[variable] = str(value)[:-1].strip()
+            elif variable in META_UB_MATRIX:
+                metadata[variable] = re.findall(r"[-+]?\d*\.\d+|\d+", str(value))
+
+        if "#data" in line:
+            # this is the end of metadata and the start of data section
+            break
+
+    # read data
+    measurements = {}
+    if data_type == ".ccl":
+        decimal = list()
+
+        if metadata["zebra_mode"] == "bi":
+            ccl_first_line = CCL_FIRST_LINE_BI
+        elif metadata["zebra_mode"] == "nb":
+            ccl_first_line = CCL_FIRST_LINE_NB
+        ccl_second_line = CCL_SECOND_LINE
+
+        for line in fileobj:
+            d = {}
+
+            # first line
+            measurement_number, *params = line.split()
+            for param, (param_name, param_type) in zip(params, ccl_first_line):
+                d[param_name] = param_type(param)
+
+            decimal.append(bool(Decimal(d["h_index"]) % 1 == 0))
+            decimal.append(bool(Decimal(d["k_index"]) % 1 == 0))
+            decimal.append(bool(Decimal(d["l_index"]) % 1 == 0))
+
+            # second line
+            next_line = next(fileobj)
+            params = next_line.split()
+            for param, (param_name, param_type) in zip(params, ccl_second_line):
+                d[param_name] = param_type(param)
+
+            d["om"] = np.linspace(

```

Comment:
@JakHolzer , could you confirm that this is correct, especially when the number of measurements is odd

Reply:
I believe so, I've tested that now on some model data and it seems everything is in order. The omega angle in the first line (not the array) is centre of the the scan and doesent have to be in the omega array. Ive tried it on odd and even numbers and mean of omega array is equal to omega value. 
for pints in 20, 21:
    omega  = 31.4
    step = 0.1

    om = np.linspace(float(omega) - (int(pints) / 2) * float(step),
        float(omega) + (int(pints) / 2) * float(step),
        int(pints))

    print(om)
    print(np.mean(om))

gives:
[30.4        30.50526316 30.61052632 30.71578947 30.82105263 30.92631579
 31.03157895 31.13684211 31.24210526 31.34736842 31.45263158 31.55789474
 31.66315789 31.76842105 31.87368421 31.97894737 32.08421053 32.18947368
 32.29473684 32.4       ]
31.4
[30.35  30.455 30.56  30.665 30.77  30.875 30.98  31.085 31.19  31.295
 31.4   31.505 31.61  31.715 31.82  31.925 32.03  32.135 32.24  32.345
 32.45 ]
31.4

In odd, the omega is middle value, in even case the omega itself is not there. Or am I missing something?

Reply:
OK, I didn't know how this "om" array should be constructed at all. This is clearly dependent on how daq works, so if you know that this array is calculated this way, then it's all good.